src/rust/inetstack/protocols/tcp/tests/simulator.rs

```diff
@@ -861,11 +861,17 @@ impl Simulation {
         } else {
             None
         };
-        let mut segment: TcpSegment =
-            TcpSegment::new(ipv4_hdr, tcp_hdr, data, false).expect("Should be able to create TCP segment");
+        let mut segment: TcpSegment = TcpSegment::new(
+            self.remote_sockaddr.ip(),
+            self.local_sockaddr.ip(),
+            tcp_hdr,
+            data,
+            false,
+        )
+        .expect("Should be able to create TCP segment");
         let mut pkt: DemiBuffer = segment.take_body().unwrap();
-
-        // Actually prepend L2 header.
+        let payload_bytes: usize = pkt.len();
```

Comment:
This should be renamed to `num_payload_bytes` or something similar. For a sec I thought these are the actual bytes. :)

Reply:
ah, good point, will do